### ISSUE TITLE
YouTube Embeds

### ISSUE THREAD

**Issue by loregamer** [cinnyapp/cinny#2036]
### Describe the problem

Can you guys make it so YouTube embeds use an iframe and displayed like this?
![image](https://github.com/user-attachments/assets/faf402c4-4078-494d-8b64-4985803aafb7)

### Describe the solution you'd like

I've made a userscript to make this possible, but would be nice if built int
`
(function () {
  "use strict";

  function fixYouTubeEmbeds() {
    const brokenEmbeds = document.querySelectorAll(
      'div.prxiv40 img[alt*="YouTube"]'
    );
    brokenEmbeds.forEach((img) => {
      let embedContainer = img.closest("div.prxiv40");
      if (embedContainer) {
        const link = embedContainer.querySelector(
          'a[href*="youtube.com/watch"]'
        );
        if (link) {
          const url = new URL(link.href);
          const videoId = url.searchParams.get("v");
          if (videoId) {
            // Create new embed
            const embed = document.createElement("iframe");
            embed.src = `https://www.youtube.com/embed/${videoId}`;
            embed.width = "480";
            embed.height = "270";
            embed.frameBorder = "0";
            embed.allow =
              "accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture";
            embed.allowFullscreen = true;

            // Replace content of the embed container
            embedContainer.innerHTML = "";
            embedContainer.appendChild(embed);

            // Remove all classes and set minimal styling
            embedContainer.className = "";
            embedContainer.style.display = "flex";
            embedContainer.style.justifyContent = "center";
            embedContainer.style.alignItems = "center";
          }
        }
      }
    });
  }

  // Run the function on page load
  window.addEventListener("load", fixYouTubeEmbeds);

  // Run the function when new messages are added
  const observer = new MutationObserver(fixYouTubeEmbeds);
  observer.observe(document.body, { childList: true, subtree: true });
})();

`

**Comment by notramo** [comment]
Definitely wouldn't be nice if built-in. I recommend to use Discord if you want embedded tracker in your messaging app. Cinny is meant to be private.

**Comment by loregamer** [comment]
> Definitely wouldn't be nice if built-in. 

> Cinny is meant to be private.

![image](https://github.com/user-attachments/assets/14493372-9259-4d04-b717-faed39c87b56)

**Comment by Handrail9** [comment]
> Definitely wouldn't be nice if built-in. I recommend to use Discord if you want embedded tracker in your messaging app. Cinny is meant to be private.

I agree with this guys sentiment even if a bit caviler. Maybe make it a toggle to turn embeds on and off, ot better yet allow users to customize posted links to redirect to private front ends. Google doesnt need my IP address, but if other people want to make that decision for themselves I say let them.
